themes/bootstrap3/js/common.js

```diff
@@ -15,21 +15,41 @@ var VuFind = (function VuFind() {
   var _elementBase;
   var _iconsCache = {};
 
-  // Emit a custom event
-  // Recommendation: prefix with vf-
-  var emit = function emit(name, detail) {
-    if (typeof detail === 'undefined') {
-      document.dispatchEvent(new Event(name));
-    } else {
-      var event = document.createEvent('CustomEvent');
-      event.initCustomEvent(name, true, true, detail); // name, canBubble, cancelable, detail
-      document.dispatchEvent(event);
+  // Event controls
+
+  let listeners = {};
+  function unlisten(event, fn) {
+    const index = listeners[event].indexOf(fn);
```

Comment:
Do you need to worry about `listeners[event]` being undefined? Maybe there should be an early return in that situation to avoid errors. (I realize in theory it shouldn't happen, but in practice maybe it could).